gotooltest/setup.go

```diff
@@ -36,13 +81,14 @@ type testContext struct {
 // It checks go command can run, but not that it can build or run
 // binaries.
 func Setup(p *testscript.Params) error {
-	var c testContext
-	if err := c.init(); err != nil {
-		return err
+	goEnv.once.Do(initGoEnv)
+	if goEnv.err != nil {
```

Comment:
Hmm, doesn't this mean that we don't be able to use the testscript command at all if there's no go command available? I wonder if if this should fail softly and just not set the relevant environment variables if the go command isn't around.

Reply:
As discussed, we will only call `gotooltest.Setup` from `cmd/testscript` if `exec.LookPath("go")` succeeds. That, plus some docs to that effect, feels like the "best" thing for now.